packages/stores/src/queries/pools/cfmm-cl-link.ts

```diff
@@ -1,28 +1,73 @@
 import { KVStore } from "@keplr-wallet/common";
-import { ChainGetter, ObservableChainQuery } from "@keplr-wallet/stores";
-import { makeObservable } from "mobx";
+import {
+  ChainGetter,
+  ObservableChainQuery,
+  ObservableChainQueryMap,
+} from "@keplr-wallet/stores";
+import { computed, makeObservable } from "mobx";
 import { computedFn } from "mobx-utils";
 
+import { ObservableQueryNodeInfo } from "../tendermint/node-info";
 import { MigrationRecords } from "./types";
 
 export class ObservableQueryCfmmConcentratedPoolLinks extends ObservableChainQuery<MigrationRecords> {
-  constructor(kvStore: KVStore, chainId: string, chainGetter: ChainGetter) {
+  // BACKWARDS COMPATIBLE QUERIES FOR PRE V17:
+  // TODO: can be safely removed after migration to v17, est Aug 22 '23
+  protected backCompatQueryCfmmToCL: ObservableQueryCfmmToConcentratedLiquidityPoolLinks;
+  protected backCompatQueryClToCfmm: ObservableQueryConcentratedLiquidityToCfmmPoolLinks;
+
+  constructor(
+    kvStore: KVStore,
+    chainId: string,
+    chainGetter: ChainGetter,
+    protected readonly queryNodeInfo: ObservableQueryNodeInfo
+  ) {
     super(
       kvStore,
       chainId,
       chainGetter,
       "/osmosis/gamm/v1beta1/cfmm_concentrated_pool_links"
     );
 
+    this.backCompatQueryCfmmToCL =
+      new ObservableQueryCfmmToConcentratedLiquidityPoolLinks(
+        kvStore,
+        chainId,
+        chainGetter
+      );
+
+    this.backCompatQueryClToCfmm =
+      new ObservableQueryConcentratedLiquidityToCfmmPoolLinks(
+        kvStore,
+        chainId,
+        chainGetter
+      );
+
     makeObservable(this);
   }
 
+  protected canFetch(): boolean {
+    return Boolean(this.queryNodeInfo.response);
+  }
+
+  @computed
+  protected get isV17() {
+    if (this.queryNodeInfo.isDevelopmentEnv) return true;
+
+    return (this.queryNodeInfo.nodeVersion ?? 0) >= 17;
+  }
+
   /** Link to concentrated liquidity pool from the given balancer/cfmm pool ID.
    *  Returns `undefined` if not loaded, `false`, if nonexistent, or the CL pool ID if it exists. */
   readonly getLinkedConcentratedPoolId = computedFn(
     (cfmmPooLId): string | false | undefined => {
       if (!this.response) return;
```

Comment:
What do you think about adding a guard to check if queryNodeInfo has a response? I believe this could prevent unexpected calls

```suggestion
      if (!this.response || this.canFetch()) return;
```

Reply:
Are you noticing unexpected calls? My thought was, the canFetch would start to return true once the node query lands. Then the relevant link query can become observed and start.

Reply:
Oh I see. Looking at the edgenet preview, it appears the v17 query is not running at all.

Reply:
On non-v17, I notice some extra calls occurring, but I believe this is unrelated to the comment I made. Here's the reference image: 
![image](https://github.com/osmosis-labs/osmosis-frontend/assets/21092519/7ad2e2ab-d010-42a5-a731-315bae1e4065).

In my comment, I want to prevent an additional call from being made if the node version query takes too long to resolve or fails to resolve altogether.